### PR TITLE
Added functionality to associate existing slack team to gluon team.

### LINKED ISSUE #4: Associate existing Slack channel as a new team's channel
screen shot 2018-02-05 at 13 38 06

Once `Use an existing channel` is selected:

* Prompt member to type the name of an existing Slack channel they want to use as their Team channel
* Add the Atomist bot automatically to the existing channel
* Link the existing channel in Gluon for that team
* Send message to the team channel with next steps etc.

### PR DESCRIPTION
**Pull request by kbristow**
- Moved common code to a seperate function and tried to do a little clean up on the promise return pyramid.
- TeamSlackChannel.ts file can still be cleaned up

Resolves #4

### CODE REVIEW

File: src/gluon/team/TeamSlackChannel.ts [hunk 1]
```diff
@@ -89,95 +93,128 @@ export class NewTeamSlackChannel implements HandleCommand {
...
+    public teamName: string;
+
+    @Parameter({
+        description: "team channel name",
+        required: true,
+    })
+    public teamChannel: string;
+
+    public handle(ctx: HandlerContext): Promise<HandlerResult> {
+
```
Comment (donovanmuller) [inline src/gluon/team/TeamSlackChannel.ts]: Remove blank line for consistency.

**Comment by kbristow** [comment]
Issue added for creating tests. See #23

**Comment by kbristow** [comment]
Command flow seen below:

Have existing slack channel:
existing team

Atomist command conversation (manually invoked in the below situation):
screen shot 2018-02-08 at 08 43 56

Atomist bot is added to existing team:
screen shot 2018-02-08 at 08 44 13

**Comment by donovanmuller** [comment]
Can you please create a follow on issue so that manually entering the team name is changed to a menu selection. The teams that this member is part of should be displayed in that menu for selection.

**Comment by kbristow** [comment]
Yes that is correct. See here:

https://github.com/absa-subatomic/quantum-mechanic/blob/6dce0507b4147d8c87f47c3b388b9f16a9884542/src/gluon/team/TeamSlackChannel.ts#L52

**Pull request merged**